src/inmanta/deploy/scheduler.py

```diff
@@ -628,13 +643,42 @@ async def report_resource_state(
                             deploying=set(),
                         )
 
+    async def _get_last_non_deploying_state_for_dependencies(
+        self, resource: ResourceIdStr
+    ) -> dict[ResourceIdStr, const.ResourceState]:
+        """
+        Get resource state for every dependency of a given resource from the scheduler state.
+        The state is then converted to const.ResourceState.
+
+        Should only be called under scheduler lock.
+
+        :param resource_id: The id of the resource to find the dependencies for
+        """
+        requires_view: Mapping[ResourceIdStr, Set[ResourceIdStr]] = self._state.requires.requires_view()
+        dependencies: Set[ResourceIdStr] = requires_view.get(resource_id, set())
+        dependencies_state = {}
+        for dep_id in dependencies:
+            new_state = self._state.resource_state[dep_id]
```

Comment:
I think we should type this. And I'm not too fond of the name `new_state` which implies a value change in the current method, what about `resource_state_object` / `resource_state_instance` ? Not a strong opinion either way, feel free to ignore.
```suggestion
            new_state: ResourceState = self._state.resource_state[dep_id]
```

Reply:
I changed it to `resource_state_object` thanks for the suggestion